doc/leak_tracking/DETECT.md

```diff
@@ -0,0 +1,165 @@
+
+### Instrument more disposables
+
+To instrument a disposable class for leak tracking, you need to dispatch object creation and disposal events to leak tracker.
+Use [this example](https://github.com/flutter/flutter/pull/141526/files) as a guide.
+
+### See leaks in a running Flutter application
+
+TODO(polina-c): implement and test this scenario
+
+1. Add [leak_tracker](https://pub.dev/packages/leak_tracker) to `dependencies` in `pubspec.yaml`.
+
+2. Before `runApp` invocation, enable leak tracking, and connect
+the Flutter memory allocation events:
+
+```dart
+import 'package:flutter/foundation.dart';
+import 'package:leak_tracker/leak_tracker.dart';
+
+...
+
+enableLeakTracking();
+FlutterMemoryAllocations.instance
+      .addListener((ObjectEvent event) => dispatchObjectEvent(event.toMap()));
+runApp(...);
+
+```
+
+3. Run the application in debug mode and watch for a leak related warnings.
+
+TODO(polina-c): add example of the warning.
+
+## Limitations
+
+### By environment
+
+At this time, leak tracking is supported only for unit tests within Flutter packages.
+
+With the latest version of leak tracker, leak tracking is not supported for:
+
+1. Web platform
+2. Running applications
+3. Pure Dart packages
+
+### By leak types
+
+Leak tracking for not-GCed leaks is experimental and is off by default.
+At this time, it is recommended to track only not-disposed leaks.
+
+### By tracked classes
+
+The leak tracker will catch leaks only for instrumented
+objects (see [concepts](CONCEPTS.md) for details).
+
+However, the good news is:
+
+1. Disposables in the Flutter Framework are instrumented.
+If your Flutter app manages widgets in a way that results in leaks,
+`leak_tracker` will catch them.
+
+2. If a leak involves at least one instrumented object,
+the leak will be caught and all
+other objects, even non-instrumented, will stop leaking as well.
+
+See [the instrumentation guidance](#instrument-your-code).
+
+### By build mode
+
+The leak tracker availability differs by build modes.
+See [Dart build modes](https://github.com/dart-lang/site-www/issues/4436)
+or [Flutter build modes](https://docs.flutter.dev/testing/build-modes).
+
+**Dart development and Flutter debug**
+
+Leak tracking is fully available.
+
+**Flutter profile**
+
+Leak tracking is available, but `FlutterMemoryAllocations` that listens to
+Flutter instrumented objects,
+should be [turned on](https://github.com/flutter/flutter/blob/15af81782e19ebe7273872f8b07ac71df4e749f2/packages/flutter/lib/src/foundation/memory_allocations.dart#L13)
+if you want to track Flutter Framework objects.
+
+**Dart production and Flutter release**
+
+Leak tracking is disabled.
+
+NOTE: If you are interested in enabling leak tracking for release mode, please, comment [here](https://github.com/dart-lang/leak_tracker/issues/25).
```

Comment:
There should probably be documentation somewhere in here about how to make sure it _doesn't_ affect release mode (e.g. what constants to use to compile code out).

Reply:
If we see a lot of interest, we will invest into discussion more. For now, there are some comments on the issue. And current focus is on regression testing.